docs/devel/api-conventions.md

```diff
@@ -263,6 +264,15 @@ This rule maintains the invariant that all JSON/YAML keys are fields in API obje
 
 Some fields will have a list of allowed values (enumerations). These values will be strings, and they will be in CamelCase, with an initial uppercase letter. Examples: "ClusterFirst", "Pending", "ClientIP".
 
+#### Unions
+
+Sometimes, at most one of a set of fields can be set.  For example, the [volumes] field of a PodSpec has 17 different volume type-specific
+fields, such as `nfs` and `iscsi`.  All fields in the set should be [Optional](#optional-vs-required).
```

Comment:
Considered saying something about that, but decided judgement can be used.  If you have like 3 options, then it is not too hard to say "at most one of A, B or C should be set".  It gets awkward with 17 options, but that is more the exceptional case.